airbyte-integrations/connectors/source-bigquery/source_bigquery/schema_helpers.py

```diff
@@ -0,0 +1,103 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from copy import deepcopy
+from typing import Any, Dict
+
+from airbyte_cdk.models import AirbyteStream
+from airbyte_cdk.models.airbyte_protocol import DestinationSyncMode, SyncMode
+
+logger: logging.Logger = logging.getLogger("airbyte")
+
+
+class SchemaTypes:
+
+    string: Dict = {"type": ["null", "string"]}
+
+    number: Dict = {"type": ["null", "number"]}
+
+    boolean: Dict = {"type": ["null", "boolean"]}
+
+    array: Dict = {"type": ["null", "array"]}
+
+    object: Dict = {"type": ["null", "object"]}
+
+
+# https://docs.airbyte.com/integrations/sources/bigquery
+SIMPLE_BIGQUERY_TYPES: Dict = {
+    "BOOL": SchemaTypes.boolean,
+    "INT64": SchemaTypes.number,
+    "FLOAT64": SchemaTypes.number,
+    "NUMERIC": SchemaTypes.number,
+    "BIGNUMERIC": SchemaTypes.number,
+    "INTEGER": SchemaTypes.number,
+    "STRING": SchemaTypes.string,
+    "BYTES": SchemaTypes.string,
+    "DATE": SchemaTypes.string,
+    "DATETIME": SchemaTypes.string,
+    "TIMESTAMP": SchemaTypes.string,
+    "TIME": SchemaTypes.string,
+    "GEOGRAPHY": SchemaTypes.string
+}
+
+COMPLEX_BIGQUERY_TYPES: Dict = {
+    "ARRAY": SchemaTypes.array,
+    "RECORD": SchemaTypes.array,
+    "STRUCT": SchemaTypes.object,
+    "JSON": SchemaTypes.object
+}
+
+
+class SchemaHelpers:
+    @staticmethod
+    def clean_name(name_str: str) -> str:
+        return name_str.replace(" ", "_").lower().strip()
+
+    @staticmethod
+    def get_json_schema(table: Dict[str, Any]) -> Dict[str, str]:
+        properties: Dict = {
+            "_bigquery_table_id": SchemaTypes.string,
+            "_bigquery_created_time": SchemaTypes.string
+        }
+
+        fields: Dict = table.get("schema", {})["fields"]
+
+        for field in fields:
+            name: str = field.get("name")
+            original_type: str = field.get("type")
+            # mode: str = field.get("mode")
+            # field_type: str = field.get("type")
+            if original_type in SIMPLE_BIGQUERY_TYPES.keys():
+                properties.update(**{name: deepcopy(SIMPLE_BIGQUERY_TYPES.get(original_type))})
+            elif original_type in COMPLEX_BIGQUERY_TYPES.keys():
+                complex_type = deepcopy(COMPLEX_BIGQUERY_TYPES.get(original_type))
+                if original_type == "ARRAY" or original_type == "RECORD":
```

Comment:
The documentation mentions `Use of RECORD/STRUCT indicates that the field contains a nested schema.` [source](https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#TableFieldSchema) so is there a reason why we treat those differently?

Reply:
Fixed here
https://github.com/artefactory/airbyte/blob/0437ac48aee0e20b2542131f7e30048f61099e73/airbyte-integrations/connectors/source-bigquery/source_bigquery/schema_helpers.py#L47